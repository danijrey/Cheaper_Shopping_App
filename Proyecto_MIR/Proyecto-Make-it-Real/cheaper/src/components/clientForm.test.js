import React from 'react';
import {render, getAllByTestId} from '@testing-library/react';
import ClientForm from './clientForm';


describe ('ClientForm',() =>{
    it ('should render correctly', ()=>{
    const {container} = render (<ClientForm/>); 
    expect (container.firstChild).toMatchSnapshot();
    });

    it('Should render form with data-testid="ClientForm"', ()=>{
        const {getByTestId}=render(<ClientForm/>);
        const client=getByTestId("ClientForm");
        expect(client).toBeInTheDocument();
    })
    it('Should render the following text ="Ingresa, Enviar"', ()=>{
        const {getByText}=render(<ClientForm/>);
        const clientText=getByText(/Ingresa/, /Enviar/);
        expect(clientText).toBeInTheDocument();
    })
});