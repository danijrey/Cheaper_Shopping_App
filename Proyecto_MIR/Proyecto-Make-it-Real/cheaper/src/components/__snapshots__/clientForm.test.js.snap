// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientForm should render correctly 1`] = `
<div
  class="App"
>
  <div
    class="FormContainer"
  >
    <img
      alt="Logo"
      class="NavBar-Logo-Client"
      src="logo.png"
    />
    <div
      class="PaginaNew-Client"
    >
      <h1
        class="TitleForm"
      >
        Ingresa tus datos
      </h1>
      <div
        class="InputsNew-Client"
      >
        <form
          class="ClientForm"
          data-testid="ClientForm"
        >
          <input
            class="Name"
            name="name"
            placeholder=" Nombre"
            type="text"
            value=""
          />
          <input
            class="LastName"
            name="lastname"
            placeholder=" Apellidos"
            type="text"
            value=""
          />
          <input
            class="Correo"
            name="clientEmail"
            placeholder=" Correo"
            type="email"
            value=""
          />
          <input
            class="Password"
            name="password"
            placeholder="ContraseÃ±a"
            type="password"
            value=""
          />
          <button
            class="SubmitClient"
            type="submit"
          >
            Enviar
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
